eitprocessing/parameters/tidal_impedance_variation.py

```diff
@@ -0,0 +1,158 @@
+            "mean": np.mean,
+            "standard deviation": np.std,
+            "median": np.median,
+        },
+    )
+    summary_stats_pixel: dict = field(
+        default_factory=lambda: {
+            "values": lambda v: v,
+            "mean": lambda v: np.mean(v, axis=0),
+            "standard deviation": lambda v: np.std(v, axis=0),
+            "median": lambda v: np.median(v, axis=0),
+        },
+    )
+    breath_detection_kwargs: dict = field(default_factory=dict)
+
+    def __post_init__(self) -> None:
+        pass
+
+
+    def compute_global_parameter(self, sequence: Sequence, data_label: str, tiv_method = 'inspiratory') -> dict | list[dict]:
+        """Compute the tidal impedance variation per breath.
+
+        Args:
+            sequence: the sequence containing the data.
+            data_label: the label of the continuous data in the sequence to determine the TIV of.
+            tiv_method: the label of which part of the breath the TIV should be determined on (inspiratory, expiratory or mean)
+        """
+        if self.method != "extremes":
+            msg = f"Method {self.method} is not implemented."
+            raise NotImplementedError(msg)
+
+        continuousdata = sequence.continuous_data[data_label]
+        eitdata = next(filter(lambda x: isinstance(x, EITData), continuousdata.derived_from))
+        data = continuousdata.values
+
+        bd_kwargs = self.breath_detection_kwargs.copy()
+        bd_kwargs["sample_frequency"] = eitdata.framerate
+        breath_detection = BreathDetection(**bd_kwargs)
+        breaths = breath_detection.find_breaths_1d(data)
+
+        start_indices, middle_indices, end_indices = zip(*breaths, strict=True)
+        start_indices = list(start_indices)
+        middle_indices = list(middle_indices)
+        end_indices = list(end_indices)
+
+        if tiv_method == 'inspiratory':
+            end_inspiratory_values = data[middle_indices]
+            start_inspiratory_values = data[start_indices]
+            tiv_values = end_inspiratory_values - start_inspiratory_values
+
+        if tiv_method == 'expiratory':
+            start_expiratory_values = data[middle_indices]
+            end_expiratory_values = data[end_indices]
+            tiv_values = start_expiratory_values - end_expiratory_values
+
+        if tiv_method == 'mean':
+            start_inspiratory_values = data[start_indices]
+            end_inspiratory_values = data[middle_indices]
+            end_expiratory_values = data[end_indices]
+            tiv_values = end_inspiratory_values - [np.mean(k) for k in zip(start_inspiratory_values, end_expiratory_values)]
+
+        result = {}
+        for name, function in self.summary_stats_global.items():
+            result[name] = function(tiv_values)
+
+        result["peak indices"] = middle_indices
+
+        return result
+
+
+    def compute_pixel_parameter(self, sequence: Sequence, data_label: str, tiv_method = 'inspiratory') -> dict | list[dict]:
```

Comment:
It seems to me that this function is for a large part a copy of `compute_global_parameter()`. Is it possible to extract the part(s) that are duplicated into separate functions that are used by these functions?